File: src/main/java/sds/offre/tache_metier/ConsulterUneFormule.java
package sds.offre.tache_metier;

import sds.offre.concept_metier.Formule;
import sds.offre.concept_metier.FormuleRepository;
import sds.offre.concept_metier.IdFormule;

public class ConsulterUneFormule {

    private FormuleRepository formuleRepository;

    public ConsulterUneFormule(FormuleRepository formuleRepository) {
        this.formuleRepository = formuleRepository;
    }

    public Formule consulte(IdFormule idFormule) {
        return formuleRepository.get(idFormule);
    }

}

File: src/main/java/sds/offre/tache_metier/ConsulterUneFormule.kt
package sds.offre.tache_metier

import sds.offre.concept_metier.Formule
import sds.offre.concept_metier.FormuleRepository
import sds.offre.concept_metier.IdFormule

class ConsulterUneFormule(private val formuleRepository: FormuleRepository) {

    fun consulte(idFormule: IdFormule): Formule {
        return formuleRepository[idFormule]
    }

}
